### PR TITLE
chore: warn if docker secret cleanup fails

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
Closes https://github.com/coder/envbuilder/issues/159

### CODE REVIEW

File: envbuilder.go [hunk 1]
```diff
@@ -604,9 +604,14 @@ func Run(ctx context.Context, options Options) error {
...
-		err = os.Remove(filepath.Join(MagicDir, "config.json"))
-		if err != nil && !os.IsNotExist(err) {
-			return fmt.Errorf("remove docker config: %w", err)
+		c := filepath.Join(MagicDir, "config.json")
+		err = os.Remove(c)
+		if err != nil {
+			if !os.IsNotExist(err) {
+				return fmt.Errorf("remove docker config: %w", err)
+			} else {
+				fmt.Fprintln(os.Stderr, "failed to remove the Docker config secret file: %w", c)
```
Comment (johnstcn) [inline envbuilder.go:613]: Why not use `options.Logger` here and warn?

Comment (BrunoQuaresma) [reply envbuilder.go:613]: I saw both options being used in the code so I didn't know what to use, to be honest.

File: envbuilder.go [hunk 2]
```diff
@@ -604,9 +604,14 @@ func Run(ctx context.Context, options Options) error {
 
 	// Remove the Docker config secret file!
 	if options.DockerConfigBase64 != "" {
-		err = os.Remove(filepath.Join(MagicDir, "config.json"))
-		if err != nil && !os.IsNotExist(err) {
-			return fmt.Errorf("remove docker config: %w", err)
+		c := filepath.Join(MagicDir, "config.json")
+		err = os.Remove(c)
```
[the remark this answers is not in the document]
Comment (johnstcn) [reply envbuilder.go:608]: Let's keep it out of scope of this PR then.

File: envbuilder.go (hunk 1 above)
Comment (johnstcn) [reply envbuilder.go:613]: That's a fair point. I'll raise a separate PR to standardize this.

**Comment by johnstcn** [comment]
I think this might also be a good place to add a check in the integration tests that the docker config.json was removed, something like this might work:

```
		statOutput := execContainer(t, ctr, "stat "+filepath.Join(envbuilder.MagicDir, "config.json"))
		require.Contains(t, statOutput, "No such file or directory")
```

**Review by johnstcn — APPROVED** [review]

**Pull request merged**